python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -455,7 +467,7 @@ def _weight_loader_impl(
             )
 
         # Flashinfer assumes w31 format for w13_weight. Same for the scales.
-        if getattr(self, "use_flashinfer_trtllm_moe", False):
+        if should_use_flashinfer_trtllm_moe():
```

Comment:
We should cache the `should_use_flashinfer_trtllm_moe` result to avoid invoking it repeatedly. Also, update other areas accordingly.